### ISSUE TITLE
$localize is not a function with Angular 14 build

### ISSUE THREAD

**Issue by softsimon** [ng-bootstrap/ng-bootstrap#4346]
### Bug description:
After upgrading to Angular 14, the Ngb Pagination component breaks with the error `TypeError: $localize is not a function`, but only in production build, it works in dev-mode.

This seems to be cause by some incompatibility of the @angular/localize library between the main build and this third party component build.

It's like this polyfill doesn't have any affect anymore:
`import '@angular/localize/init';`

### Versions of Angular, ng-bootstrap and Bootstrap:

Tried with ng-bootstrap 13.0.0-beta.1 and 12.x

Angular:
14.0.5
ng-bootstrap:
13.0.0-beta.1
Bootstrap:
4.5

**Comment by rissom** [comment]
I think I have the same problem. I upgraded to Bootstrap 5, but that didn't help. It works on my PC (development), but throws an error on the server (production, at runtime on some pages of my application. Could be the ones with the pagination component, I didn't check yet).

**Comment by rissom** [comment]
Update: I solved some of my problems by updating my package.json from "popper.js": "^1.16.1" to "@popperjs/core": "^2.10.2" (and running "npm install" of course. Unfortunately I still have some pages where I get the "TypeError: $localize is not a function" - error.

**Comment by St1c** [comment]
same issue here

**Comment by rissom** [comment]
Halleluja! I found my problem. It was a hidden file called ".browserlistrc". I don't think I put it there. Anyway, removing it solved it (for me).

**Comment by St1c** [comment]
I don't think that's the solution .browserlistrc is essential file for angular compiler. It says to the compiler which browsers should be supported, and then the compiler compiles Javascript files in compliance with that configuration.

**Comment by rissom** [comment]
Of course. But since deleting it solved the problem for me, the bug seems to be specific to a certain Javascript configuration. I am fine just staying with the default and will not pursue it further. My .browserslistrc file contained: 


\# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
\# For additional information regarding the format and rule options, please see:
\# https://github.com/browserslist/browserslist#queries

\# You can see what browsers were selected by your queries by running:
\#   npx browserslist

\> 0.5%
last 2 versions
Firefox ESR
not dead
not IE 9-11 # For IE 9-11 support, remove 'not'.

**Comment by St1c** [comment]
Following discussions here: https://github.com/angular/angular/issues/46503 on the same issue:

It seems that if you put "not ios_saf 12.2-12.6" to the end of .browserlistrc , then Angular compiles just fine. 

I spent quite some time debugging it, the problem seems to be, that for legacy browsers compiler creates this compiled javascript code: 

`const i18n_1 = $localize(_t2 || (_t2 = _2`:@@ngb.alert.close:Close`));`

In the compiled code _2 is defined as:

`let _2 => t => t;`

which causes problems, as $localize function should be used as tag function for template literals, where it expects messageParts: TemplateStringsArray as argument, but here it receives plain string. Removing " ios_saf 12.2-12.6" from browser compatibility list translates all custom IDs just fine.

**Comment by Alderman8** [comment]
I had the same issue in my project but with ngbTimePicker and ngbDatePicker, adding `not ios_saf 12.2-12.6` to _.browserliststrc_ really helped, thank you @St1c

**Comment by softsimon** [comment]
Ok duplicate of https://github.com/angular/angular/issues/46503

**Comment by dopsun** [comment]
I discovered this ticket when I am facing same issue.

TL;DR;
In my case, I eventually fixed my problem, by moving `messages.xlf` generated to the default location, which is project root, instead of `src/locale`.

*A longer version*
The issue happens, while I'm generating `messages.xlf` with
```bash
ng extract-i18n --output-path src/locale
```

Issue disappeared by deleting generated file, and regenerate with
```bash
ng extract-i18n --output-path src/locale
```

The only difference in my case, is location of generated file.
